tests/unit/algo/hooks/test_saliency_map_processing.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+import numpy as np
+import pytest
+import torch
+from otx.algo.utils.xai_utils import process_saliency_maps, process_saliency_maps_in_pred_entity
+from otx.core.config.explain import ExplainConfig
+from otx.core.data.entity.classification import MulticlassClsBatchPredEntityWithXAI, MultilabelClsBatchPredEntityWithXAI
+from otx.core.types.explain import TargetExplainGroup
+
+NUM_CLASSES = 5
+BATCH_SIZE = 3
+RAW_SIZE = 7
+OUT_SIZE = 224
+
+PRED_LABELS = [[0, 2, 3], [1], []]
+PRED_LABELS_TOP_ONE = [[1], [0], [4]]
+SALIENCY_MAPS = [np.ones((NUM_CLASSES, RAW_SIZE, RAW_SIZE), dtype=np.uint8) for _ in range(BATCH_SIZE)]
+SALIENCY_MAPS_IMAGE = [np.ones((RAW_SIZE, RAW_SIZE), dtype=np.uint8) for _ in range(BATCH_SIZE)]
+
+
+@pytest.mark.parametrize("postprocess", [False, True])
+def test_process_all(postprocess) -> None:
+    explain_config = ExplainConfig(target_explain_group=TargetExplainGroup.ALL, postprocess=postprocess)
+
+    with pytest.raises(ValueError, match="Shape mismatch."):
+        processed_saliency_maps = process_saliency_maps(SALIENCY_MAPS_IMAGE, explain_config, PRED_LABELS)
```

Comment:
```suggestion
@pytest.fixture
def fxt_saliency_maps():
    return [np.ones((NUM_CLASSES, RAW_SIZE, RAW_SIZE), dtype=np.uint8) for _ in range(BATCH_SIZE)]

@pytest.fixture
def fxt_saliency_maps_image():
    return [np.ones((RAW_SIZE, RAW_SIZE), dtype=np.uint8) for _ in range(BATCH_SIZE)]

@pytest.mark.parametrize("postprocess", [False, True])
def test_process_all(postprocess, fxt_saliency_maps) -> None:
    explain_config = ExplainConfig(target_explain_group=TargetExplainGroup.ALL, postprocess=postprocess)

    with pytest.raises(ValueError, match="Shape mismatch."):
        processed_saliency_maps = process_saliency_maps(fxt_saliency_maps, explain_config, PRED_LABELS)
```

Please use pytest fixture to follow the common practice and avoid to use global variables which are hard to debug if they are silently changed somewhere.